### PR TITLE
Display menu and menu item

### PR DESCRIPTION
**Pull request by ghost**
## PR #3 - Display menu and menu item
[Link to Pivital Tracker user story](https://www.pivotaltracker.com/story/show/167007659)

Admittedly, we have made one stylistic error on this pull request.  There are two tasks in this PR, each as a separate feature within a single story. The first feature was scripted in the 'development' branch while the second feature was scripted in the '01_display_menu_item' branch.  We unfortunately made a lot of progress and then realized in retrospect.

That said, there are two tasks in this PR.
1. The first task is visiting the landing page and seeing the company name, the menu title, and one food category.
2. The second task uses FactoryBot to create a test instance of a menu item to procedurally display the menu item attributes on the menu.  
3. We also decided to include a seed file because it aided with the creation of the factory bot for the feature test and we will be using a seed file for eventual population of the final menu.

### CODE REVIEW

File: features/visitor_menu.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Feature: Visitor can see the menu
```
Comment (oliverochman) [inline features/visitor_menu.feature]: Remove this test, we only need one feature file

File: features/display_menu_item.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Feature: Visitor can see a prodcut on the menu
+    As a visitor,
+    In order to get an overview of the restaurant's products,
+    I would like to be able to see a product on the menu.
+
+    Background:
+        Given the following menu item exist
+            | name         | description                                                                             | price   | animal |
+            | Kitty Heaven | Fresh, seared yellow tail tuna, served on a bed of catnip picked daily from our garden. | 140     | cat    |
```
Comment (oliverochman) [inline features/display_menu_item.feature:9]: I would like to see multiple MenuItems in this feature

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  # get 'menu/index'
```
Comment (oliverochman) [inline config/routes.rb]: you can remove this, old syntax

Comment (oliverochman) [reply config/routes.rb]: never push up commented out code if you are not in the middle of implementing a feature

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,7 @@
...
+
+  resources :menu_items
+
+  root 'menu#index'
```
Comment (oliverochman) [inline config/routes.rb]: please refactor this to the new syntax

e.g.
```
root controller: :menu_items, action: :index
```

File: app/models/menu_item.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class MenuItem < ApplicationRecord
+    validates :name, presence: true
```
Comment (oliverochman) [inline app/models/menu_item.rb]: please refactor this

E.g
`validates_presence_of :name, :description, :price`

File: app/controllers/menu_controller.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class MenuController < ApplicationController
+  def index
+    @menu_items = MenuItem.all
+  end
+
+  def create
```
Comment (oliverochman) [inline app/controllers/menu_controller.rb]: the create action is out of scope for this feature, please remove it

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/visitor_menu.feature (hunk 1 above)
Comment (ghost) [reply features/visitor_menu.feature]: Incorporated into display_menu_item.feature

File: config/routes.rb (hunk 3 above)
Comment (ghost) [reply config/routes.rb]: Comment cleaned.

File: config/routes.rb (hunk 4 above)
Comment (ghost) [reply config/routes.rb]: Syntax updated.

File: app/models/menu_item.rb (hunk 5 above)
Comment (ghost) [reply app/models/menu_item.rb]: Refactored with modern syntax.

File: app/controllers/menu_controller.rb (hunk 6 above)
Comment (ghost) [reply app/controllers/menu_controller.rb]: Removed until later feature.

File: features/display_menu_item.feature (hunk 2 above)
Comment (ghost) [reply features/display_menu_item.feature:9]: Two products are now in the background for the feature test.

**Review by oliverochman — APPROVED** [review]

**Comment by tochman** [comment]
Great communication guys. Looks good.

**Pull request merged**